experimental/generated/beats/fields.ecs.yml

```diff
@@ -5262,7 +5262,11 @@
       description: 'Domain of the url, such as "www.elastic.co".
 
         In some cases a URL may refer to an IP and/or port directly, without a domain
-        name. In this case, the IP address would go to the `domain` field.'
+        name. In this case, the IP address would go to the `domain` field.
+
+        If the URL contains a literal IPv6 address, `domain` may include the `[` and
```

Comment:
My preference would be for stronger wording (`shall` over `may`) to encourage consistency. Something along the lines of

> If the original URL contains a literal IPv6 address enclosed in brackets then `url.domain` shall include those brackets.

Reply:
Yeah I agree with the stronger wording. We haven't been using "shall" much if at all, in ECS. Perhaps we stick with "must"?

Reply:
The next strongest wording would be `should`, which indicates a strong preference, but won't, for example, start treating things as invalid if it isn't met.

Reply:
I wanted to avoid inadvertently leading anyone to alter the value by adding brackets if missing in the original (albeit an unlikely scenario).

Perhaps something like:

> If the URL contains a literal IPv6 address enclosed by `[` and `]` (IETF RFC 2732), the `[` and `]` characters must also be captured in the `domain` field.

Reply:
++ on not leading people to alter the value, if the IPv6 is not enclosed in brackets.

I think @ebeahan's wording suggested above works well. I'm also fine with using the word "should", as suggested by @sgryphon.